output/cloud/expv2/output.go

```diff
@@ -19,6 +21,18 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// RequestMetadatasCollector is an interface for collecting request metadatas
+// and retrieving them so they can be flushed using a Flusher.
+type RequestMetadatasCollector interface {
+	CollectRequestMetadatas([]metrics.SampleContainer)
+	PopAll() insights.RequestMetadatas
+}
+
+// Flusher is an interface for flushing data to the cloud.
+type Flusher interface {
+	Flush(ctx context.Context) error
+}
+
```

Comment:
```suggestion
// RequestMetadatasCollector is an interface for collecting request metadatas
// and retrieving them so they can be flushed using a Flusher.
type requestMetadatasCollector interface {
	CollectRequestMetadatas([]metrics.SampleContainer)
	PopAll() insights.RequestMetadatas
}

// Flusher is an interface for flushing data to the cloud.
type flusher interface {
	flush(ctx context.Context) error
}

```
I'm going to open a PR for unexporting all the methods/interfaces/fields not required to be exported. So let's not add more. It sounds they are used only internally to the current package.

Reply:
Fixed in [dd1741f](https://github.com/grafana/k6/pull/3100/commits/dd1741f49285edbdabd289356b360932dc862beb).